core/src/test/java/fj/IOTest.java

```diff
@@ -0,0 +1,34 @@
+package fj;
+
+import fj.data.IO;
+import fj.data.IOFunctions;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+
+public class IOTest {
+
+    String result = null;
+
+    @Test
+    public void testLift() throws IOException {
+        final IO<String> readName = () -> new BufferedReader(new StringReader("foo")).readLine();
+        final F<String, IO<Unit>> upperCaseAndPrint = F1Functions.<String, IO<Unit>, String>o(this::println).f(String::toUpperCase);
+        final IO<Unit> readAndPrintUpperCasedName = IOFunctions.bind(readName, upperCaseAndPrint);
+        readAndPrintUpperCasedName.run();
+        assertThat(result, is("FOO"));
+    }
+
+    public IO<Unit> println(final String s) {
```

Comment:
I know.  I thought it would be better to return `IO<String>`and return s here.  Then in the method above you can replace result with `readAndPrintUpper....` and remove the variable result.  This would improve referential transparency.